.evergreen/config_generator/components/integration.py

```diff
@@ -34,7 +34,8 @@
     ('debian12', None, ['Release'], ['shared',         ], [      20], [None], ['plain',        ], [False, True], ['latest'], ['single',                     ]),
     ('debian12', None, ['Release'], ['shared',         ], [None,   ], [None], [         'csfle'], [False,     ], ['latest'], [          'replica', 'sharded']),
 
-    ('macos-1100', None, ['Release'], ['shared', 'static'], [None], [None], ['plain'], [False, True], ['5.0', 'latest'], ['single']),
+    ('macos-14',       None, ['Release'], ['shared', 'static'], [None], [None], ['plain'], [False, True], ['5.0',         ], ['single']),
```

Comment:
I am unsure of the user impact for dropping support of macOS 11, 12, and 13. macOS 13 is [still supported](https://endoflife.date/macos). I filed CXX-3193 to request dropping macOS 11 and 12.

Suggest documenting support for macOS 11 and 12 is deprecated in CHANGELOG.md to give users a heads up. Restore one macOS 11 task to ensure the driver continues to build on macOS 11.

Reply:
Jumping from MacOS 11 to MacOS 14 distros is motivated by [DEVPROD-8869](https://jira.mongodb.org/browse/DEVPROD-8869) and [DEVPROD-8870](https://jira.mongodb.org/browse/DEVPROD-8870). I do not think our test coverage needs to be strictly 1-to-1 with platforms we (do not) document as actively supported. LMK if we nevertheless want to maintain coverage of MacOS 11 (+ perhaps 13; there is no distro for 12) until we are forced to move away once the distros are decommissioned.

Reply:
Restored some MacOS 11 coverage (integration matrix, latest only) after further discussion. Included a deprecation notice in the changelog as suggested. An attempt to migrate from `macos-11` to `macos-11-arm64` (as done in https://github.com/mongodb/mongo-c-driver/pull/1791) was blocked by [DEVPROD-13222](https://jira.mongodb.org/browse/DEVPROD-13222).